src/uu/seq/src/floatparse.rs

```diff
@@ -0,0 +1,198 @@
+// This file is part of the uutils coreutils package.
+//
+// For the full copyright and license information, please view the LICENSE
+// file that was distributed with this source code.
+// spell-checker:ignore extendedbigdecimal bigdecimal hexdigit numberparse
+use crate::extendedbigdecimal::ExtendedBigDecimal;
+use crate::number::PreciseNumber;
+use crate::numberparse::ParseNumberError;
+use bigdecimal::BigDecimal;
+use num_traits::FromPrimitive;
+
+pub fn parse_hexadecimal_float(s: &str) -> Result<PreciseNumber, ParseNumberError> {
+    let value = parse_float(s)?;
+    let number = BigDecimal::from_f64(value).ok_or(ParseNumberError::Float)?;
+    let fractional_digits = i64::max(number.fractional_digit_count(), 0) as usize;
+    Ok(PreciseNumber::new(
+        ExtendedBigDecimal::BigDecimal(number),
+        0,
+        fractional_digits,
+    ))
+}
+
+fn parse_float(s: &str) -> Result<f64, ParseNumberError> {
+    let mut s = s.trim();
+
+    // Detect a sign
+    let sign = if s.starts_with('-') {
+        s = &s[1..];
+        -1.0
+    } else if s.starts_with('+') {
+        s = &s[1..];
+        1.0
+    } else {
+        1.0
+    };
+
+    // Is HEX?
+    if s.starts_with("0x") || s.starts_with("0X") {
```

Comment:
I think this function can be simplified
For example:

    // Validate and strip HEX prefix
    if !s.starts_with("0x") && !s.starts_with("0X") {
        return Err(ParseNumberError::Float);
    }
    s = &s[2..];

Reply:
Agree. Thx